src/gval/comparison/pairing_functions.py

```diff
@@ -223,12 +222,15 @@ class PairingDict(dict):
         Value to use instead of np.nan.
     """
 
-    # TODO: Perhaps this could be np.finfo(float).max
-    replacement_value = "NaN"
-
     def __init__(self, *args, **kwargs):
         # make a tmp dict with
         tmp_dict = dict(*args, **kwargs)
+        dtype = type(np.array(list(tmp_dict.keys()))[0][0])
+
+        if issubclass(dtype, np.integer):
+            self.replacement_value = np.iinfo(dtype).max
```

Comment:
Just curious if the user pass `np.iinfo(dtype).max` and `np.nan` if it would cause a collision?